browser/web_discovery/web_discovery_tab_helper.cc

```diff
@@ -45,8 +45,10 @@ WebDiscoveryTabHelper::WebDiscoveryTabHelper(content::WebContents* contents)
 
 WebDiscoveryTabHelper::~WebDiscoveryTabHelper() = default;
 
-void WebDiscoveryTabHelper::PrimaryPageChanged(content::Page& page) {
-  if (page.GetMainDocument().GetLastCommittedURL() != GURL(kBraveSearchUrl))
+void WebDiscoveryTabHelper::DidFinishLoad(
+    content::RenderFrameHost* render_frame_host,
+    const GURL& validated_url) {
+  if (validated_url != GURL(kBraveSearchUrl))
```

Comment:
do we need `GURL(...)`? maybe make `kBraveSearchUrl` as `base::StringPiece`? or we can create `base::StringPiece` right here:
```
if (validated_url != base::StringPiece(kBraveSearchUrl))
```
it will be more efficient.

Reply:
Fixed.